src/runtime_src/core/pcie/emulation/sw_emu/generic_pcie_hal2/shim.cxx

```diff
@@ -446,39 +446,26 @@ namespace xclswemuhal2
 
         if (xilinxInstall.empty())
         {
-          std::cerr << "ERROR : [SW-EM 10] Please make sure that the XILINX_VITIS environment variable is set correctly" << std::endl;
+          std::cerr << "ERROR : [SW_EMU 10] Please make sure that the XILINX_VITIS environment variable is set correctly" << std::endl;
           exit(1);
         }
 
-        std::string modelDirectory("");
-#if defined(RDIPF_aarch64)
-        if (boost::filesystem::exists(xilinxInstall + "/data/emulation/unified/sw_emu/zynqu/model/genericpciemodel"))
-          modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/zynqu/model/genericpciemodel";
-        else
-          modelDirectory = xilinxInstall + "/data/emulation/unified/cpu_em/zynqu/model/genericpciemodel";
-
-#elif defined(RDIPF_arm64)
-        if (boost::filesystem::exists(xilinxInstall + "/data/emulation/unified/sw_emu/zynq/model/genericpciemodel"))
-          modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/zynq/model/genericpciemodel";
-        else
-          modelDirectory = xilinxInstall + "/data/emulation/unified/cpu_em/zynq/model/genericpciemodel";
-#else
-        if (boost::filesystem::exists(xilinxInstall + "/data/emulation/unified/sw_emu/generic_pcie/model/genericpciemodel"))
-          modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/generic_pcie/model/genericpciemodel";
-        else
-          modelDirectory = xilinxInstall + "/data/emulation/unified/cpu_em/generic_pcie/model/genericpciemodel";
-#endif
+        std::string modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/generic_pcie/model/genericpciemodel";
 
         FILE *filep;
         if ((filep = fopen(modelDirectory.c_str(), "r")) != NULL)
         {
           // file exists
+          if (access(modelDirectory.c_str(), X_OK) != 0)
+          {
+            std::cout << "genericpciemodel binary does not have executable permission." << std::endl;
```

Comment:
If executable permissions are not present then don't you think it needs to exit here?

Reply:
Yes, please check, updated as per suggestion.

Reply:
checked.